sources.go

```diff
@@ -102,3 +105,61 @@ func (s pstateSource) Discover() ([]string, error) {
 
 	return features, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Network Source
+
+// Implements main.FeatureSource.
+type networkSource struct{}
+
+func (s networkSource) Name() string { return "network" }
+
+// reading the network card details from sysfs and determining if SR-IOV is enabled for each of the network interfaces
+func (s networkSource) Discover() ([]string, error) {
+	features := []string{}
+	netInterfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, fmt.Errorf("can't obtain the network interfaces details: %s", err.Error())
+	}
+	// iterating through network interfaces to obtain their respective number of virtual functions
+	for _, netInterface := range netInterfaces {
+		if strings.Contains(netInterface.Flags.String(), "up") && !strings.Contains(netInterface.Flags.String(), "loopback") {
+			totalVfsPath := "/sys/class/net/" + netInterface.Name + "/device/sriov_totalvfs"
+			totalBytes, err := ioutil.ReadFile(totalVfsPath)
+			if err != nil {
+				stderrLogger.Printf("SR-IOV not supported for network interface: %s: %s", netInterface.Name, err.Error())
+				continue
+			}
+			total := bytes.TrimSpace(totalBytes)
+			t, err := strconv.Atoi(string(total))
+			if err != nil {
+				stderrLogger.Printf("Error in obtaining total number of virtual functions for network interface: %s: %s", netInterface.Name, err.Error())
+				continue
+			}
+			if t > 0 {
+				stdoutLogger.Printf("%d total number of virtual functions on network interface: %s", t, netInterface.Name)
+				features = append(features, "sriov")
+				numVfsPath := "/sys/class/net/" + netInterface.Name + "/device/sriov_numvfs"
+				numBytes, err := ioutil.ReadFile(numVfsPath)
+				if err != nil {
+					stderrLogger.Printf("SR-IOV not supported for network interface: %s: %s", netInterface.Name, err.Error())
```

Comment:
I'm just trying to delineate between `support` and `configuration` in the logs. `sriov_totalvfs` determines if there is support and `sriov_numvfs` determines if sriov is configured. We should use `not configured` here since we are checking for `sriov_numvfs`.